chore: upgrade deps to prep for 0.2.0-rc1 release

kentbull opened the pull request:
This does a few things:
1. bumps version to 0.3.0 for the next release.
2. Upgrades to KERI 1.2.2 (only had to change the `getFelItemPreIter` return tuple)
3. Adds a smaple KERIA config at `scripts/keri/cf/keria.json` with sample tock configuration.
4. Updates all dependencies to the latest versions in PyPi.
5. Uses specific versions in setup.py so builds are strictly repeatable.
6. adds a little more project metadata

iFergal commented:
@kentbull OK, this will supersede #345 - thanks for the extra changes.

However, we haven't released `0.2.0` yet - we're still on dev tags. Hence, I think this PR should upgrade to `0.2.0-rc1` and create a release branch. And then a separate PR can upgrade to `0.3.0-dev0`.

iFergal commented on scripts/keri/cf/keria.json:
See https://github.com/WebOfTrust/keria/commit/912e68144ee0f31536c77d67f732ee82469d210c

@lenkan has changed it to use the env instead of this file.

kentbull replied:
The keria.json file is still used though, and the tocks are only configurable with the JSON file right now.
It would be worth talking over which approach we like better. The `keria.json` file was intended to be a bootstrap config file for all of the configuration needs of an agent.

And this file is just to make sure the example keria.json file is complete and correct.

iFergal replied:
OK fair, it's just an example and not used by the Dockerfile. That's OK - I just wanted to avoid having both and it being confusing. Tocks should probably be configurable by the env too then.

Aside from that, the env can bootstrap the configuration now in the same way.

Probably should be documented.

kentbull replied:
I'll add docs for this.

iFergal commented on setup.py:
```diff
@@ -34,10 +34,11 @@
     license='Apache Software License 2.0',
     description='KERIA: KERI Agent in the cloud',
-    long_description="KERIA: KERI Agent in the cloud.",
+    long_description=open("README.md").read(),
```
Is this a standard practice, and is it OK with all the images? (I guess so with `text/markdown`)

kentbull replied:
It is a nice way to get the Readme to show up on the PyPi package page.
Here's the [docs](https://packaging.python.org/en/latest/guides/making-a-pypi-friendly-readme/) on how to do it. I found a better way using [this SO answer](https://stackoverflow.com/a/69655159/2084253).

iFergal commented on setup.py:
```diff
@@ -64,34 +65,33 @@
     project_urls={
+        'Documentation': 'https://weboftrust.github.io/keridoc',
         'Issue Tracker': 'https://github.com/WebOfTrust/keria/issues',
+        'Source': 'https://github.com/WebOfTrust/keria/issues',
```
This URL is for issue tracker

kentbull replied:
whoops, good catch, copy paste error.

iFergal commented on setup.py:
```diff
@@ -34,10 +34,11 @@
 setup(
     name='keria',
-    version='0.2.0-dev6',  # also change in src/keria/__init__.py
+    version='0.3.0',  # also change in src/keria/__init__.py
```
Already left a comment on the version itself, but you're missing the change to the `Makefile`

kentbull replied:
This was changed to 0.2.0-rc1 in the most recent commit (force pushed).

kentbull commented:
@iFergal sounds good to me, i's now changed to `0.2.0-rc1`. I also updated versions in the build pipeline and added the most recent Alpine image.

To slightly optimize image size I swapped out alpine-sdk for gcc. Only the gcc dependency is needed in order to properly link libsodium. If it weren't for linking libsodium then we wouldn't need gcc, which was the entire reason alpine-sdk was included.

iFergal reviewed (approved):
LGTM. Not sure if you're still going to update any docs on the env vs config file but not critical.

2byrds reviewed (approved):
LGTM

kentbull commented:
@iFergal see the docs I added in the PR. You can `cd docs/ && make html && make open` to view the docs. I changed the theme to the ReadTheDocs theme which is nice on the eyes.

iFergal reviewed (approved):
LGTM!
